metpy/calc/tools.py

```diff
@@ -1024,11 +1027,13 @@ def first_derivative(f, **kwargs):
     delta_slice0[axis] = slice(None, 1)
     delta_slice1[axis] = slice(1, 2)
 
-    combined_delta = delta[delta_slice0] + delta[delta_slice1]
-    big_delta = combined_delta + delta[delta_slice0]
-    left = (- big_delta / (combined_delta * delta[delta_slice0]) * f[slice0] +
-            combined_delta / (delta[delta_slice0] * delta[delta_slice1]) * f[slice1] -
-            delta[delta_slice0] / (combined_delta * delta[delta_slice1]) * f[slice2])
+    combined_delta = delta[tuple(delta_slice0)] + delta[tuple(delta_slice1)]
```

Comment:
Ugh...this is ugly enough for me to want to make something to hide this absurdity.

Reply:
Would a pair of `_convert_to_tuple(*args)` and `_convert_to_list(*args)` helpers be a good approach? Or do you have an idea for another approach?

Reply:
I'm thinking something more OO, and possibly more automatic. I don't have anything particular in mind at the moment, but need to play with some ideas.